### PR TITLE
[AIRFLOW-3284] Azure Batch AI Operator (#18)

### PR DESCRIPTION
**Pull request by elizabethhalper**
### Jira

- [ ] My PR addresses the following [Airflow Jira](https://issues.apache.org/jira/browse/AIRFLOW/) issues and references them in the PR title:
  -https://issues.apache.org/jira/browse/AIRFLOW-3284
  - In case you are fixing a typo in the documentation you can prepend your commit with [AIRFLOW-3284], code changes always need a Jira issue.

### Description

- [ ] Here are some details about my PR, including screenshots of any UI changes:
Adds an operator to create an Azure Batch AI cluster. There is a hook, example DAG, and tests to support this operator. Azure Batch AI allows users to experiment and train deep learning and AI models in parallel at scale.
Operator supports creating a batch AI cluster by using any Marketplace image when given the publisher, offer, sku, and version.

### Tests

- [ ] My PR adds the following unit tests __OR__ does not need testing for this extremely good reason:
There are unit tests that test a successful execution of the operator and an execution that raises an Airflow exception.

### Commits

- [ ] My commits all reference Jira issues in their subject lines, and I have squashed multiple commits if they address the same issue. In addition, my commits follow the guidelines from "[How to write a good git commit message](http://chris.beams.io/posts/git-commit/)":
  1. Subject is separated from body by a blank line
  1. Subject is limited to 50 characters (not including Jira issue reference)
  1. Subject does not end with a period
  1. Subject uses the imperative mood ("add", not "adding")
  1. Body wraps at 72 characters
  1. Body explains "what" and "why", not "how"

### Documentation

- [ ] In case of new functionality, my PR adds documentation that describes how to use it.
  - When adding new operators/hooks/sensors, the autoclass documentation generation needs to be added.
  - All the public functions and the classes in the PR contain docstrings that explain what it does

### Code Quality

- [ ] Passes `flake8`

### CODE REVIEW

File: airflow/contrib/example_dags/example_gcp_sql.py [hunk 1]
```diff
@@ -29,22 +29,32 @@
 """
 
 import os
-import datetime
```
Comment (ashb) [inline airflow/contrib/example_dags/example_gcp_sql.py:32]: I think something funny has gone on with your PR here - there are many unrelated changes.

Could you try rebasing (or closing and opening a new PR if you prefer) to get just your Azure related changes?

Comment (Fokko) [reply airflow/contrib/example_dags/example_gcp_sql.py:32]: Looking at the conflicts, something went wrong. Maybe cherry-pick your commit on top of a clean master?

Comment (elizabethhalper) [reply airflow/contrib/example_dags/example_gcp_sql.py:32]: Thank you. I will close this PR and create a new one with the correct files.
